lib/core/src/wallet.rs

```diff
@@ -407,6 +407,9 @@ impl OnchainWallet for LiquidOnchainWallet {
                 }
                 Err(e) => Err(e.into()),
             };
+
+        self.persister.set_last_scanned_derivation_index(index)?;
```

Comment:
This is going to store last_derivation_index + 5, meaning if the data-sync receives an increased index by 1-5 it won't force the sync

Reply:
Yes, but I think that is fine. The stored index is the one we did end up scanning to. Do you see this causing any issues?

Reply:
My point is an incremented derivation index received by the data-sync will not trigger a full scan. Which I think is half the reason for the PR, no?

Reply:
@danielgranhao I think what Ross means is that we should save the index we get from the persister here: https://github.com/breez/breez-sdk-liquid/pull/775/files#diff-6b184d7d7439883bb4f686575e75810a483be9041940320c97a094c81f0b077cR386
and not the one we increment by 5 (to scan a bit forward). Because eventually we compare the persister value for the decision to scan: https://github.com/breez/breez-sdk-liquid/pull/775/files#diff-28794968ead61e86665263d494740e1a8fbe62fcfed42ed568bcf77fd51ca31dR438

Reply:
I've changed the persisted index to not include the buffer in [de5bbf0](https://github.com/breez/breez-sdk-liquid/pull/775/commits/de5bbf05eefb965ad9c622ba03f0bdb159b2c2e7). Let's be on the safe side.